### ISSUE TITLE
Listing ssh keys fails if TPM only supports one session

### ISSUE THREAD

**Issue by laanwj** [irtimmer/tpm2-pk11#65]
My AMD laptop's TPM (kernel driver tpm_crb) only supports one session at a time.

This causes the following to fail:
```
ssh-keygen -D libtpm2-pk11.so
C_OpenSession failed: 5
cannot read public key from pkcs11
```

I checked in a debugger and the reason for this is that `/dev/tpm0` is opened twice - once in `C_Initialize` (which works) and another time in `C_OpenSession` (which fails).

Not sure at which level this problem would need to be resolved, if possible at all.
(I've, for now, hacked around this in a very ugly way)

**Comment by irtimmer** [comment]
Fixed with commit 2f0ed78f33d09b401f2392b34c12dae0152e5519

**Comment by laanwj** [comment]
Thanks!

**Comment by forx617** [comment]
Hi guys,

problem seems to be reappeared.

I'm using:
1. ARM platform linux based with a running real /dev/tpm0 device
2. last released tpm2-tss stack : [2.0.0 release](https://github.com/tpm2-software/tpm2-tss/releases)
3. last commit for tpm2-pk11 : 2343865fb5b1749317c8a160166d6c3ccbc3fa50

Created a persistent tpm key:

> tpm2_listpersistent 
> persistent-handle[0]:0x81000000 key-alg:rsa hash-alg:sha256 object-attr:fixedtpm|fixedparent|sensitivedataorigin|userwithauth|restricted|decrypt

Then the following fail:

> ssh-keygen -D /usr/lib/pkcs11/libtpm2-pk11.so 
> ERROR:tcti:src/tss2-tcti/tcti-device.c:281:Tss2_Tcti_Device_Init() Failed to open device file /dev/tpm0: Device or resource busy 
> C_OpenSession failed: 5
> cannot read public key from pkcs11
